src/gtfs/mod.rs

```diff
@@ -412,13 +412,18 @@ fn remove_stop_zones(model: Model) -> Result<Collections> {
         .retain(|sp| sp.stop_type != StopType::Zone);
     let stop_point_ids: Vec<&Idx<StopPoint>> =
         collections.stop_points.get_id_to_idx().values().collect();
-    let mut vjs = collections.vehicle_journeys.take();
-    for vj in &mut vjs {
-        vj.stop_times
-            .retain(|st| stop_point_ids.contains(&&st.stop_point_idx));
-    }
-    vjs.retain(|vj| !vj.stop_times.is_empty());
-    collections.vehicle_journeys = CollectionWithId::new(vjs)?;
+
+    let vjs_to_keep = collections
+        .vehicle_journeys
+        .into_iter()
+        .filter(|vj| {
+            vj.stop_times
+                .iter()
+                .all(|st| stop_point_ids.contains(&&st.stop_point_idx))
+        })
+        .collect();
+
+    collections.vehicle_journeys = CollectionWithId::new(vjs_to_keep)?;
```

Comment:
You can directly use `.retain()` function here keeping the same closure (and removing a `&` if you made the modification in the comment above :wink:).

```suggestion
    collections.vehicle_journeys.retain(|vj| {
        vj.stop_times
            .iter()
            .all(|st| stop_point_ids.contains(&st.stop_point_idx))
    });
```

Reply:
done